gateway/middleware.go

```diff
@@ -66,17 +69,21 @@ func (tr TraceMiddleware) ProcessRequest(w http.ResponseWriter, r *http.Request,
 		setContext(r, ctx)
 		return tr.TykMiddleware.ProcessRequest(w, r, conf)
 	} else if baseMw := tr.Base(); baseMw != nil {
-		if cfg := baseMw.Gw.GetConfig(); cfg.OpenTelemetry.Enabled {
-			ctx, span := baseMw.Gw.TracerProvider.Tracer().Start(r.Context(), tr.Name())
-			defer span.End()
-
-			setContext(r, ctx)
+		cfg := baseMw.Gw.GetConfig()
+		if cfg.OpenTelemetry.Enabled {
+			var span tyktrace.Span
+			if baseMw.Spec.DetailedTracing {
+				var ctx context.Context
+				ctx, span = baseMw.Gw.TracerProvider.Tracer().Start(r.Context(), tr.Name())
+				defer span.End()
+				setContext(r, ctx)
+			} else {
+				span = tyktrace.SpanFromContext(r.Context())
```

Comment:
Move this function to otel internal pkg, so we don't have to import /opentelemetry/trace here